### ISSUE TITLE
new window opening is delayed on X11 with multiple displays

### ISSUE THREAD

**Issue by duhdugg** [wez/wezterm#3300]
### What Operating System(s) are you seeing this problem on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

bspwm 0.9.10

### WezTerm version

20230320-124340-559cb7b0

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I am getting delayed window spawns when launching new windows only while connected to my docking station which has three external displays.

When not connected to my docking station, new windows open instantaneously.
When connected to my docking station, there is a delay of about 0.5 seconds before the window opens. Sometimes it takes up to 1 second before the window opens. The log I'm attaching shows 529 milliseconds between `TermWindow::new_window` being called and the screens being logged within the `resolve_geometry` function in `window/src/connection.rs`.

### To Reproduce

This should be reproducible on any X11 system with multiple displays, as I believe the issue is due to the nature of how `xcb::Connection` works. The more screens you have connected to the system, the more you're calling `XConnection::send_and_wait_request`, and more time is ultimately spent waiting on `xcb::Connection::wait_for_reply` to return.

### Configuration

I have tested that the issue persists with no config.

### Expected Behavior

Instant window opening is ideal.

### Logs

```
16:46:48.163  TRACE  wezterm_gui::termwindow            > TermWindow::new_window called with mux_wind
ow_id 0 TerminalSize { rows: 24, cols: 80, pixel_width: 800, pixel_height: 480, dpi: 96 } Dimensions 
{ pixel_width: 800, pixel_height: 480, dpi: 96 }
16:46:48.164  TRACE  wezterm_gui::termwindow            > RequestedWindowGeometry { width: Pixels(800
.0), height: Pixels(480.0), x: None, y: None, origin: ScreenCoordinateSystem }
16:46:48.262  TRACE  async_io::driver                   > block_on()
16:46:48.262  TRACE  async_io::driver                   > block_on: sleep until notification
16:46:48.692  TRACE  window::connection                 > Screens { main: ScreenInfo { name: "DP-2-2"
, rect: Rect(2560x1440 at (0, 0)), scale: 1.0, max_fps: Some(60) }, active: ScreenInfo { name: "DP-2-
2", rect: Rect(2560x1440 at (0, 0)), scale: 1.0, max_fps: Some(60) }, by_name: {"DP-2-1": ScreenInfo 
{ name: "DP-2-1", rect: Rect(2048x1152 at (2560, 0)), scale: 1.0, max_fps: Some(60) }, "DP-2-2": Scre
enInfo { name: "DP-2-2", rect: Rect(2560x1440 at (0, 0)), scale: 1.0, max_fps: Some(60) }, "DP-2-3": 
ScreenInfo { name: "DP-2-3", rect: Rect(2560x1440 at (4608, 0)), scale: 1.0, max_fps: Some(60) }}, vi
rtual_rect: Rect(7168x1440 at (0, 0)) }
```

### Anything else?

I'll be submitting a PR that is related to this shortly.

**Comment by wez** [comment]
Does the `xrandr` utility also stall for a similar period? If so, that's the source of your issue.
See also: https://github.com/wez/wezterm/issues/2073

**Comment by duhdugg** [comment]
yes it does

**Comment by wez** [comment]
I'm going to close this as a duplicate of #2073; it's a local system/configuration issue at its root and isn't something that should be compensated for in wezterm.
